src/libraries/System.Security.Cryptography.Xml/tests/CanonicalXmlEntityReferenceTest.cs

```diff
@@ -0,0 +1,170 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Text;
+using System.Xml;
+using Xunit;
+
+namespace System.Security.Cryptography.Xml.Tests
+{
+    public class CanonicalXmlEntityReferenceTest
+    {
+        [Fact]
+        public void EntityReferenceInCanonicalization()
+        {
+            // This test exercises CanonicalXmlEntityReference by using a transform
+            // that internally creates a CanonicalXmlDocument and loads XML into it.
+            // When an XmlNodeReader reads from a document containing entity references,
+            // the target document's CreateEntityReference method is called.
+
+            string xml = @"<!DOCTYPE doc [
+<!ENTITY test ""TestValue"">
+]>
+<root>&test;</root>";
+
+            XmlDocument doc = new XmlDocument();
+            doc.PreserveWhitespace = true;
+            doc.LoadXml(xml);
+
+            // Use C14N transform which internally uses CanonicalXmlDocument
+            XmlDsigC14NTransform transform = new XmlDsigC14NTransform();
+            transform.LoadInput(doc);
+            
+            Stream output = (Stream)transform.GetOutput();
+            string result = new StreamReader(output, Encoding.UTF8).ReadToEnd();
+            
+            // Entity should be expanded in canonical form
+            Assert.Contains("TestValue", result);
+            Assert.Contains("<root>", result);
+        }
+
+        [Fact]
+        public void EntityReferenceWithXmlNodeList()
+        {
+            // Test with XmlNodeList input which triggers different code path in CanonicalXml
+            // When using XmlNodeList, nodes not in the list are not included, which may affect
+            // entity reference expansion in the canonical output
+            string xml = @"<!DOCTYPE doc [
+<!ENTITY test ""Hello"">
+]>
+<root><child>&test;</child></root>";
+
+            XmlDocument doc = new XmlDocument();
+            doc.PreserveWhitespace = true;
+            doc.LoadXml(xml);
+
+            XmlNodeList nodeList = doc.SelectNodes("//child")!;
+
+            XmlDsigC14NTransform transform = new XmlDsigC14NTransform();
+            transform.LoadInput(nodeList);
+            
+            Stream output = (Stream)transform.GetOutput();
+            string result = new StreamReader(output, Encoding.UTF8).ReadToEnd();
+            
+            // The node should be present in canonical form
+            Assert.Contains("<child>", result);
+            Assert.Contains("</child>", result);
```

Comment:
It is not obvious from these expected Contains checks that the output doesn't actually include the expanded entity.  (It also wouldn't contain bare words or descendants, since they aren't part of the xpath query)

Once again, Equals is stronger.

```suggestion
            Assert.Equals("<child></child>", result);
```